### ISSUE TITLE
Build a baseline weakly semi-supervised learning model

### ISSUE THREAD

**Issue by yenchiah** [MultiX-Amsterdam/ijmond-camera-ai#11]
Use the [UniMatch-V2](https://github.com/LiheYoung/UniMatch-V2) model as the baseline. The model should be pre-finetuned on the `smoke5k` dataset and then finetuned on `ijmond_pseudo_masks` (pseudo pixel-level labels) and `ijmond_vid` (unlabeled) datasets. The `ijmond_seg` dataset should be used for testing only.

See the following paper for the UniMatch model:
- [UniMatch-V2](https://arxiv.org/abs/2410.10777)
- [UniMatch-V1](https://arxiv.org/abs/2208.09910)

[This paper](https://arxiv.org/abs/2301.03992) inspires our baseline design, which uses two phases. Phase one creates pseudo masks from bounding boxes, uses [Segment Anything](https://github.com/facebookresearch/segment-anything). Phase two uses a semi-supervised learning pipeline, which is UniMatch-V2. 

Create a new branch to work on this issue. When it is done, create a pull request and ask for code review.

Update: on Jan 5, 2026, Yen-Chia added [a script](https://github.com/MultiX-Amsterdam/ijmond-camera-ai/blob/unimatch/bbox_learn/split_ijmond_seg.py) for splitting the `ijmond_seg` dataset into training, validation, and test sets. The logic and usage of the code is documented in the [README](https://github.com/MultiX-Amsterdam/ijmond-camera-ai/blob/unimatch/bbox_learn/README.md) file. The experiment setting is also in the README.
